developer_manual/app_publishing_maintenance/maintainer.rst

```diff
@@ -0,0 +1,33 @@
+.. _app-maintainer:
+
+===========
+Maintainers
+===========
+
+.. sectionauthor:: Christoph Wurst <christoph@winzerhof-wurst.at>
+
+The role of a Nextcloud app maintainer is to oversee all processes around an app. The responsibilities are:
+
+* Triage bug reports
+
+  + Make sure reported bugs are reproducible, ask for more info if necessary
+  + Check for duplicates
+  + Close resolved tickets
+
+* Manage pull requests
+
+  + Ensure pull request receive a review in reasonable time
+  + Merge pull requests that pass the peer review and all continuous integration checks
+  + Trigger backports if applicable
+
+* Manage releases
+
+  + Ensure pre-releases are ready for deployment in pre-production of Nextcloud server
+  + Ensure compatible releases are ready for deployments of Nextcloud server releases
+
+* Manage vulnerabilities
+
+  + Monitor vulnerable dependencies, e.g. through Github security alerts, and release updates timely
+  + Coordinate disclosed vulnerabilities with the Nextcloud security team
+
+To avoid the so called `bus factor <https://en.wikipedia.org/wiki/Bus_factor>`_, it is highly recommended to have more than one maintainer for each app. Apps can have one primary maintainer and one or more backup maintainers. For transparency, it can make sense to declare the maintainers of an app in the app's repository README file, so that others know whom to contact.
```

Comment:
Talked recently with @AndyScherzinger about the "where to put" as being able to machine/script-read the information would be quite handy so we can automatically create reviews/alert etc in 3rd places, but for that we would need a dedicated format.

Two suggestions came to mind for that:

1. Extend the author information in info.xml to allow for a github handle: https://github.com/nextcloud/appstore/blob/master/nextcloudappstore/api/v1/release/info.xsd#L307-L314
2. Use the Codeowner file https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners#codeowners-file-location and require a line for `appinfo/info.xml`

This could then be used read e.g. when we create our security reviews and automatically mention the maintainers when their app is affected by something

Reply:
I would prefer to leave this to the maintainers. If they want to set themselves as code owner in their assigned repos fine by me. We don't need to enforce that, do we?
Code owners will make most sense for apps inside the server repo

Reply:
```suggestion
To avoid the so called `bus factor <https://en.wikipedia.org/wiki/Bus_factor>`_, it is highly recommended to have more than one maintainer for each app. Apps can have one primary maintainer and one or more backup maintainers.

For transparency and to allow scripted reading of the information, it is recommended to declare the maintainers of an app in the app's repository ``.github/CODEOWNERS`` file with a specific line for ``/appinfo/info.xml``, so that others know whom to contact. See this `<notifications app pull request https://github.com/nextcloud/notifications/pull/1524/files>`_ for sample code.
```

Reply:
Goal of the codeowner's file for apps would be to bind it to the info xml to the owners get notified / assigned on PRs that bump the version of the app _and_ to have it easily / standardized machine readable for the sec report generation

Reply:
So please change from readme to codeowners

Reply:
So please change from readme to codeowners

Reply:
In my understanding we expect maintainers to see any PR, so they also see PRs that touch the version number. With the exception of new migrations, the version number will only be touched by the maintainers, no? Put differently: I'm okay if someone leverages CODEOWNERS if it eases their workflow, but I dislike making it a hard requirement.

The target of the developer documentation is any Nextcloud app in our open source community. Do we mandate these changes for all of the apps or only Nextcloud GmbH maintained ones? For the latter we can use an internal doc.

Reply:
> Do we mandate these changes for all of the apps or only Nextcloud GmbH maintained ones? For the latter we can use an internal doc.

The latter for now, yes. Security team can only contact others via email anyway.

Reply:
Another machine-readable format for the maintainer and contact info could be the info.mail.author attribute: https://github.com/nextcloud/contacts/pull/3323/commits/afcd82e254cf533f3fa98c4dad73c42ccb035ac3. If someone sets a real email address it can be used to contact them for security topics. If it's a github noreply address the user can be pinged. Just an idea.

Reply:
CODEOWNERS has been put in place for Gmbh maintained apps. Is the rest of the generic maintainer role description ready to be merged?